tmt/utils.py

```diff
@@ -4787,11 +4796,18 @@ def git_clone(
                 url, destination
                 ), env=env)
     except RunError:
+        if retry_times == 0:
+            raise
         if not shallow:
             # Do not retry if shallow was not used
             raise
         # Git server might not support shallow cloning, try again (do not modify url)
-        return git_clone(url, destination, common, env, shallow=False, can_change=False)
+        shallow = retry_times > 1
+        retry_times -= 1
```

Comment:
Probably not necessary - if the last step of the function is to call `git_clone` recursively, it can be called with `retry_times=retry_times - 1`, local change to `retry_times` affects nothing in this frame anymore.

Reply:
Do you mean to use
```patch
         return git_clone(url, destination, common, env,
                          shallow=shallow,
                          can_change=False,
-                         retry_times=retry_times)
+                         retries=retries - 1,
...<snip>...
```
?

Reply:
Correct. The value shouldn't be needed, even the `shallow` change above can be done unconditionally: if we're at this point, we already know we're trying again, which means that `if shallow is True`, we can switch it to `False` no matter how many retries are left.

Reply:
Seems to be irrelevant now when `tmt.utils.retry()` is used for retrying.